.azure/pipelines/jobs/default-build.yml

```diff
@@ -121,8 +121,11 @@ jobs:
       inputs:
         versionSpec: 10.x
   - ${{ if and(eq(parameters.installJdk, 'true'), eq(parameters.agentOs, 'Windows')) }}:
-    - powershell: ./eng/scripts/InstallJdk.ps1 '11.0.1'
+    - powershell: ./eng/scripts/InstallJdk.ps1 '11.0.1';Write-Host "##vso[task.setvariable variable=PATH;]${env:PATH};$env:JAVA_HOME";
       displayName: Install JDK 11
+  - ${{ if and(eq(parameters.installJdk, 'true'), eq(parameters.agentOs, 'Windows')) }}:
+    - powershell: Write-Host "##vso[task.setvariable variable=PATH;]${env:PATH};$env:JAVA_HOME";
+      displayName: Set JAVA Path
```

Comment:
@dougbu Can you help me out with pipelines? (I've never used them). I'm basically trying to set the PATH Correctly so that our checks for prerequisites work

Reply:
@javiercn a couple of pointers
- the [sample for Java in pipelines](https://docs.microsoft.com/en-us/azure/devops/pipelines/languages/java?view=azure-devops) doesn't attempt to update `$env:Path` globally
  - instead that sample, wraps Gradlew, Java et cetera to execute the exact command needed or (in one case) updates `$env:Path` just before using Java
- at the moment, this repo behaves similarly
  - for example, see the https://github.com/aspnet/AspNetCore/blob/master/src/SignalR/clients/java/signalr/gradlew and https://github.com/aspnet/AspNetCore/blob/master/src/SignalR/clients/java/signalr/gradlew.bat scripts

Bottom line, I suggest doing something tactical in the scripts you need.